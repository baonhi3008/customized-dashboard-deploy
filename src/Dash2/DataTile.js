import React from 'react';


const DataTile = ({children}) => (
  <h1>{children}</h1>
);

export default DataTile;