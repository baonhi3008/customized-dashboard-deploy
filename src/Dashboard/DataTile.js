import React from "react";
import styled, { css } from "styled-components";
import { SelectableTile } from "../Shared/Tile";
import { fontSize3, fontSizeBig, greenBoxShadow } from "../Shared/Styles";
import { CoinHeaderGridStyled } from "../Settings/WardHeaderGrid";
import { AppContext } from "../App/AppProvider";

const JustifyRight = styled.div`
  justify-self: right;
`;

const JustifyLeft = styled.div`
  justify-self: left;
`;

const TickerPrice = styled.div`
  ${fontSizeBig}
`;

const ChangePct = styled.div`
  color: green;
  ${props =>
    props.red &&
    css`
      color: red;
    `}
`;

const numberFormat = number => +(number + "").slice(0, 7);

const PriceTileStyled = styled(SelectableTile)`
  ${props =>
    props.compact &&
    css`
      display: grid;
      ${fontSize3}
      grid-gap: 5px;
      grid-template-columns: repeat(3, 1fr);
      justify-items: right;
    `}

  ${props =>
    props.currentFavorite &&
    css`
      ${greenBoxShadow}
      pointer-events: none;
    `}
`;

function ChangePercent({ data }) {
  return (
    <JustifyRight>
      <ChangePct red={data.CHANGEPCT24HOUR < 0}>
        {numberFormat(data.CHANGEPCT24HOUR)}%
      </ChangePct>
    </JustifyRight>
  );
}

function PriceTileTag({ sym, ward,district,data, currentFavorite, setCurrentFavorite }) {
  return (
    <PriceTileStyled
      onClick={setCurrentFavorite}
      currentFavorite={currentFavorite}
    >
      <CoinHeaderGridStyled>
        {/* <div>{sym}</div> */}
		<div>{ward}</div>
		<JustifyRight>{district}</JustifyRight>
        {/* <ChangePercent data={data} /> */}
      </CoinHeaderGridStyled>
      <TickerPrice>{numberFormat(data.PRICE)}</TickerPrice>
    </PriceTileStyled>
  );
}

function PriceTileCompactTag({
  sym,
  data,
  currentFavorite,
  setCurrentFavorite
}) {
  return (
    <PriceTileStyled
      onClick={setCurrentFavorite}
      compact
      currentFavorite={currentFavorite}
    >
      <JustifyLeft>{sym}</JustifyLeft>
      <ChangePercent data={data} />
      <div>${numberFormat(data.PRICE)}</div>
    </PriceTileStyled>
  );
}

function PriceTile({coinList,price, index }) {
  let sym = Object.keys(price)[0];
  let ward = coinList[sym]["NAME_3"]
  let district = coinList[sym]["NAME_2"]
  let data = price[sym]["USD"];
  let TileClass = index < 10 ? PriceTileTag : PriceTileCompactTag;

  return (
    <AppContext.Consumer>
      {({ currentFavorite, setCurrentFavorite }) => (
        <TileClass
          sym={sym}
          data={data}
		  ward = {ward}
		  district = {district}
          currentFavorite={currentFavorite === sym}
          setCurrentFavorite={() => setCurrentFavorite(sym)}
        />
      )}
    </AppContext.Consumer>
  );
}

export default PriceTile;
