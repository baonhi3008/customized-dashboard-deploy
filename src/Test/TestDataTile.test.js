import React from 'react';
import { shallow } from 'enzyme';
import DataTile from '../Dash2/DataTile';

const title = 'Test DataTile';

let wrapped = shallow(<DataTile>{title}</DataTile>);


describe('DefaultText', () => {
  it('should render the DefaultText Component accordingly', () => { 
expect(wrapped).toMatchSnapshot();
  });  it('renders the DefaultTexts children component', () => {
expect(wrapped.find('h1').text()).toEqual(title);
  });
});