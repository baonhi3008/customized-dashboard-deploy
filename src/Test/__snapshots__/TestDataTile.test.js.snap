// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultText should render the DefaultText Component accordingly 1`] = `
<h1>
  Test DataTile
</h1>
`;
